masterfiles#654 pkgsrc package module sometimes matches wrong package

bahamat opened the pull request:
Fixes #654, #656, #657.

```
$ echo File=mtr | ./pkgsrc get-package-data
PackageType=repo
Name=mtr
```

nickanderson commented:
[Package Module API Reference](https://docs.cfengine.com/lts/reference-standard-library-package_modules.html) 

> For repository based package names the module should not return Version and Architecture, since they are often ambiguous in repository situations, and any discrepancies will be handled at the install stage instead.

bahamat commented:
Not returning the version is a bug in the specification since multiple versions may be available, in which case the wrong version may get installed. One wonders what the get-package-data subcommand is for, except  to `| sed s/File/Name/`.

Nevertheless, I've addressed these issues as well.

nickanderson commented:
@kacf can you comment on the protocol?

bahamat commented:
It was previously discussed here: https://dev.cfengine.com/issues/7328

The workaround was that version info can be fed to `get-package-data`, but then is discarded if returned.

bahamat commented:
At any rate, how the protocol *should* work is aside from the changes in this PR. There are two bugs pending to be fixed here, and the module now conforms to the current protocol specification. I don't want to make this PR about changing the spec.

kacf commented:
Changes look right to me.
